stress/shell/src/main/java/alluxio/stress/cli/StressMasterBench.java

```diff
@@ -130,6 +135,76 @@ public void prepare() throws Exception {
     }
   }
 
+  private void deletePaths(FileSystem fs, Path basePath) throws Exception {
+    // the base dir has sub directories per task id
+    if (!fs.exists(basePath)) {
+      return;
+    }
+    FileStatus[] subDirs = fs.listStatus(basePath);
+    if (subDirs.length == 0) {
+      return;
+    }
+
+    // determine the fixed portion size
+    int fixedSize = fs.listStatus(new Path(subDirs[0].getPath(), "fixed")).length;
+
+    long batchSize = 50_000;
+    int deleteThreads = 256;
+    ExecutorService service =
+        ExecutorServiceFactories.fixedThreadPool("bench-delete-thread", deleteThreads).create();
+
+    for (FileStatus subDir : subDirs) {
+      LOG.info("Cleaning up all files in: {}", subDir.getPath());
+      AtomicLong globalCounter = new AtomicLong();
+      Path fixedBase = new Path(subDir.getPath(), "fixed");
+      long runningLimit = 0;
+
+      // delete individual files in batches, to avoid the recursive-delete problem
+      while (!Thread.currentThread().isInterrupted()) {
+        AtomicLong success = new AtomicLong();
+        runningLimit += batchSize;
+        long limit = runningLimit;
+
+        List<Callable<Void>> callables = new ArrayList<>(deleteThreads);
+        for (int i = 0; i < deleteThreads; i++) {
+          callables.add(() -> {
+            while (!Thread.currentThread().isInterrupted()) {
+              long counter = globalCounter.getAndIncrement();
+              if (counter >= limit) {
+                globalCounter.getAndDecrement();
+                return null;
+              }
+              Path deletePath;
+              if (counter < fixedSize) {
+                deletePath = new Path(fixedBase, Long.toString(counter));
+              } else {
+                deletePath = new Path(subDir.getPath(), Long.toString(counter));
+              }
+              if (fs.delete(deletePath, true)) {
+                success.getAndIncrement();
+              }
+            }
+            return null;
+          });
+        }
+        service.invokeAll(callables, 1, TimeUnit.MINUTES);
```

Comment:
Despite what the doc seems to say, invokeAll doesn't necessarily guarantee that all the futures are done. So I think here, we should be checking if all of the futures are done. Log if some are not done and shutdown/reconfigure the service on every loop.

Reply:
This is ok. If this times out, it cancels all remaining tasks. That is fine, because the next round will just create more tasks, and any remaining tasks will be taken care of the final recursive delete.